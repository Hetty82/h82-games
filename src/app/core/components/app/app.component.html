<app-layout [title]="title" [subTitle]="subTitle"></app-layout>

<app-icon-sprite></app-icon-sprite>
