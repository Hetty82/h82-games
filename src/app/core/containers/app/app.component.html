<app-layout>
  <h1>Games</h1>
  <h2>Made with Angular by hetty82</h2>
</app-layout>
