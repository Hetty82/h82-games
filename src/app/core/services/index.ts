import { GamesService } from './games.service'
import { UserService } from './user.service'

export const services: any[] = [
  GamesService,
  UserService,
]

export * from './games.service'
export * from './user.service'
