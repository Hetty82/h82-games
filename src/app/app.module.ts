import { NgModule } from '@angular/core'
import { BrowserModule } from '@angular/platform-browser'
import { HttpClientModule } from '@angular/common/http'
import { RouterModule } from '@angular/router'

import { EffectsModule } from '@ngrx/effects'
import { StoreDevtoolsModule } from '@ngrx/store-devtools'
import { StoreModule } from '@ngrx/store'
import { StoreRouterConnectingModule, RouterStateSerializer } from '@ngrx/router-store'

import { CoreModule } from './core/core.module'

import { AppComponent } from './core/components'

import * as fromStore from './store'

import { routes } from './app.routes'
import { environment } from '../environments/environment'


@NgModule({
  bootstrap: [ AppComponent ],
  imports: [
    // Angular
    BrowserModule,
    HttpClientModule,
    RouterModule.forRoot(routes),

    // Ngrx
    EffectsModule.forRoot(fromStore.coreEffects),
    StoreModule.forRoot(fromStore.reducers, { metaReducers: fromStore.metaReducers }),
    // DevTools Instrumentation must be imported after importing StoreModule
    StoreDevtoolsModule.instrument({
      name: 'h82 Games - NgRx DevTools',
      logOnly: environment.production,
    }),
    StoreRouterConnectingModule.forRoot({ stateKey: 'router' }),

    // Own
    CoreModule.forRoot(),
  ],
  providers: [
    {
      provide: RouterStateSerializer,
      useClass: fromStore.CustomRouterStateSerializer,
    },
  ],
})
export class AppModule { }
