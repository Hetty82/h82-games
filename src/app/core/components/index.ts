import { AppComponent } from './app/app.component'
import { LayoutComponent } from './layout/layout.component'
import { NotFoundComponent } from './not-found/not-found.component'

export const components: any[] = [
  AppComponent,
  LayoutComponent,
  NotFoundComponent,
]

export * from './app/app.component'
export * from './layout/layout.component'
export * from './not-found/not-found.component'
