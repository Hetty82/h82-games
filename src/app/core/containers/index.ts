import { AppComponent } from './app/app.component'
import { GamesComponent } from './games/games.component'
import { LoginComponent } from './login/login.component'
import { NotFoundComponent } from './not-found/not-found.component'

export const containers: any[] = [
  AppComponent,
  GamesComponent,
  LoginComponent,
  NotFoundComponent,
]

export * from './app/app.component'
export * from './games/games.component'
export * from './login/login.component'
export * from './not-found/not-found.component'
