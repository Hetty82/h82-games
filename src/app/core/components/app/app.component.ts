import { Component } from '@angular/core'

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.sass'],
})
export class AppComponent {
  title = 'Games'
  subTitle = 'Made with Angular by hetty82'
}
