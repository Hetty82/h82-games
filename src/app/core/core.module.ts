import { NgModule } from '@angular/core'
import { CommonModule } from '@angular/common'
import { RouterModule } from '@angular/router'

import * as fromComponents from './components'
import * as fromContainers from './containers'
import * as fromServices from './services'


@NgModule({
  imports: [
    CommonModule,
    RouterModule,
  ],
  declarations: [
    ...fromComponents.components,
    ...fromContainers.containers,
  ],
  exports: [
    fromContainers.AppComponent,
  ],
})
export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule,
      providers: fromServices.services,
    }
  }
}
