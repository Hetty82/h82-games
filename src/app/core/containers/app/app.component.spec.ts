import { NO_ERRORS_SCHEMA } from '@angular/core'
import { TestBed, async, ComponentFixture } from '@angular/core/testing'

import { AppComponent } from './app.component'


describe('AppComponent', () => {
  let component: AppComponent
  let fixture: ComponentFixture<AppComponent>

  beforeEach(async(() => {
    TestBed.configureCompiler({ preserveWhitespaces: false } as any).configureTestingModule({
      declarations: [ AppComponent ],
      schemas: [ NO_ERRORS_SCHEMA ],
    })
    .compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent)
    component = fixture.componentInstance
    fixture.detectChanges()
  })

  it('should create component', () => {
    expect(component).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot()
  })
})
