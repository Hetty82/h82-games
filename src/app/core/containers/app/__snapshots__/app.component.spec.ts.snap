// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<app-root
  title={[Function String]}
>
  <app-layout>
    <h1>
      Games
    </h1>
    <h2>
      Made with Angular by hetty82
    </h2>
  </app-layout>
</app-root>
`;
